common/src/java/org/apache/hadoop/hive/conf/HiveConf.java

```diff
@@ -5138,6 +5138,15 @@ public static enum ConfVars {
     LLAP_WEBUI_SPNEGO_PRINCIPAL("hive.llap.webui.spnego.principal", "",
         "The LLAP WebUI SPNEGO service principal. Configured similarly to\n" +
         "hive.server2.webui.spnego.principal"),
+    HIVE_SERVER2_WEBUI_HTTP_COOKIE_MAX_AGE("hive.server2.ui.http.cookie.max.age", "86400s",
+        new TimeValidator(TimeUnit.SECONDS),
+        "Maximum age in seconds for server side cookie used by HS2 in HTTP mode."),
+    HIVE_SERVER2_WEBUI_HTTP_COOKIE_DOMAIN("hive.server2.ui.http.cookie.domain", null,
+        "Domain for the HS2 generated cookies"),
+    HIVE_SERVER2_WEBUI_HTTP_COOKIE_PATH("hive.server2.ui.http.cookie.path", null,
+        "Path for the HS2 generated cookies"),
+    HIVE_SERVER2_WEBUI_ENABLE_LDAP("hive.server2.webui.enable.ldap", false,
```

Comment:
I think for making this future-proof for other authentication methods to be added, this should be extendable like:
```
hive.server2.webui.auth.enable=true/false
hive.server2.webui.auth.method=ldap
```

Reply:
Done

Reply:
thanks!